lighthouse-core/lib/network-recorder.js

```diff
@@ -168,15 +165,17 @@ class NetworkRecorder extends EventEmitter {
   }
 
   /**
-   * Construct network records from a log of devtools protocol messages.
-   * @param {!DevtoolsLog} log
+   * Construct network records from a log of devtools protocol messages. If a
+   * driver is provided, it will be used for request content lookups.
+   * @param {!DevtoolsLog} devtoolsLog
+   * @param {!Driver=} driver
```

Comment:
>IMO, audits shouldn't need to live request content if gatherers didn't save them into a separate artifact

right, it's not possible for audits to do that on current master already, as we throw away the driver by that point

> meaning it can just be a method on a driver to getResourceContent and pass in a network record id, the construction of network records is not capable of reconstructing the content anyway

well I wish I had thought of that before I spent all that time reverse engineering that ['web-inspector'](https://github.com/GoogleChrome/lighthouse/blob/d3c216d1b2b0e2fba7e1716943c94cd093b2e70d/lighthouse-core/lib/web-inspector.js#L246-L252) madness and then making it work again with this PR :)

I think Paul was going for the current approach because it makes accessing the content pretty seamless. We already use devtools' `WebInspector.NetworkRequest` objects, and they have `requestContent` on them, so it makes some sense to hook that method up similarly to how devtools does and have it send the command on your behalf.

It seems like we have a "this thing works here but doesn't work there" in both cases, so the choice would be:
1) (current behavior) Support `record.requestContent()`. Meanwhile throw on `record.requestContent()` when called when a driver wasn't used to initialize things and document that it only works in audits, not gatherers
2) If an audit wants to access content, they have to do `driver.getRequestContent(record)` or whatever. Meanwhile always throw on `record.requestContent()` and document driver method if you want content.

I do kind of like the explicitness of the second approach, making it crystal clear that you'll only be able to access content through the driver (so if you don't have a driver anymore, you won't get content). OTOH we're breaking the API of `NetworkRequest` and breaking away from the way devtools code does things. OTOOH probably no one would have known that `record.requestContent` existed without paul anyways and would have happily used a driver method (analogous to `getObjectProperty`).

@paulirish wdyt?